includes/models/model.llms.course.php

```diff
@@ -234,11 +235,15 @@ public function get_difficulty( $field = 'name' ) {
 
 		} else {
 
+			$difficulties = array();
+
 			foreach ( $terms as $term ) {
 
-				return $term->$field;
+				$difficulties[] = $term->$field;
```

Comment:
we're not doing any check on the existence of the `$term->field`, so this will raise a fatal if the requested field doesn't exist.
I think we can live with it, but in this case then we can use [`wp_list_pluck()`](https://developer.wordpress.org/reference/functions/wp_list_pluck/).

I wonder though if we should add the separator as new parameter.
I also wonder if we want to add an additional parameter like return type... array or string.
Your opinion?

Reply:
we can use `property_exist` to check if `$term->$field` exists.
Like this:

```
else {

	foreach ( $terms as $term ) {
			
			if( property_exists( $term, $field ) ) {

				$difficulties = wp_list_pluck( $terms, $field );

			}

		}

		return implode( ', ', $difficulties );
	}
}
```

> I wonder though if we should add the separator as new parameter.

parameter as in? where?

> I also wonder if we want to add an additional parameter like return type... array or string.

yeah, it would be good. Like this maybe:

` * @return string | array

Reply:
@actuallyakash 
> parameter as in? where?

Like a new parameter of the function:
`	public function get_difficulty( $field = 'name', $separator = ', '  ) {`


> yeah, it would be good. Like this maybe:

>  `* @return string | array`

Yeah like that, but then you have to return an array, otherwise what that  stays for?
The idea would be to optionally return an array instead of a string, depending on an additional parameter. But let's not do it for now.